web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,234 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/* Controllers */
+var mudrodControllers = angular.module('mudrodControllers', []);
+
+mudrodControllers.controller('searchCtrl', ['$scope', '$rootScope', '$location', 'Autocomplete',
+    function ($scope, $rootScope, $location, Autocomplete) {
+        $scope.options = {
+            preference: 'And'
+        };
+        //$scope.relatedTerms = [];
+        //$rootScope.searchOptions = {};
+        
+        $scope.complete = function(string) {  
+            $scope.hidethis = false;  
+            $scope.hideoption = true;
+            var output = [];  
+            Autocomplete.get({term: string},
+                function success(response) {
+                    //alert($scope.challenge.question);
+                    //console.log("Success:" + JSON.stringify(response));
+                    //$scope.blogList = response;
+                    $scope.filterSearch = response;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+            );  
+        }  
+
+        $scope.fillTextbox = function(string){  
+            $scope.options.term = string;  
+            $scope.hidethis = true;  
+            $scope.hideoption = false;
+        }  
+
+        $scope.search = function(options) {
+            $rootScope.searchOptions = angular.copy(options);
+            
+            $location.path("/metadataView/"+options.term);
+
+        };
+
+    }]);
+    
+mudrodControllers.controller('vocabularyCtrl', ['$scope', '$rootScope', 'VocabList',
+    function vocabularyCtrl($scope, $rootScope, VocabList) {
+
+        var word = $rootScope.searchOptions.term;
+        var opt = $rootScope.searchOptions.preference;
+        VocabList.get({query: word},
+                function success(response) {
+                    //alert($scope.challenge.question);
+                    //console.log("Success:" + JSON.stringify(response));
+                    //$scope.blogList = response;
+                    $scope.ontologyList = response.graph.ontology;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+        );
+    }]);
+
+mudrodControllers.controller('metadataViewCtrl', ['$rootScope', '$scope', '$routeParams', 'MetaData', 'PagerService',
+    function metadataViewCtrl($rootScope, $scope, $routeParams, MetaData, PagerService) {
+        var vm = this;
+        vm.PDItems = [];
+        vm.pager = {};
+        vm.setPage = setPage;
+        vm.totalMatches = 0;
+
+        var word = new String();
+        var opt = new String();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a literal value for this. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3APrimitiveWrappers)